app/src/main/java/dcom/taneja/dhruv/floater/screenshotservice/ScreenshotActivity.java

```diff
@@ -0,0 +1,134 @@
+package dcom.taneja.dhruv.floater.screenshotservice;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.v4.app.ActivityCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+import dcom.taneja.dhruv.floater.R;
+
+/**
+ * Created by PrateekGarg on 1/12/2018.
+ */
+
+public class ScreenshotActivity extends AppCompatActivity {
+    private String pathToFile = null;
+    private String TAG = "Permission Information:";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.screenshot_activity);
+        Button updateButton = (Button) findViewById(R.id.screenshotButton);
+        updateButton.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View view) {
+                updateTimer();
+                takeScreenshot();
+            }
+        });
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           String permissions[], int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+            Log.v(TAG, "Permission: " + permissions[0] + "was " + grantResults[0]);
+            //resume tasks needing this permission
+        }
+
+    }
+    private void updateTimer() {
+        TextView timer = (TextView) findViewById(R.id.editText);
```

Comment:
a new instance is created every time this function is invoked. this can better be a private field of the class